src/main/java/org/springframework/data/couchbase/core/convert/CryptoConverter.java

```diff
@@ -133,16 +124,20 @@ private byte[] coerceToBytesWrite(CouchbasePersistentProperty property, Converti
 			}
 			plainText = ja.toBytes();
 		} else if (cnvs.isSimpleType(sourceType)) { // simpleType
-			String plainString = value != null ? value.toString() : null; // TODO - this will ignore @JsonValue
+			String plainString = value != null ? value.toString() : null;
 			if ((sourceType == String.class || targetType == String.class) || sourceType == Character.class
-					|| sourceType == char.class || Enum.class.isAssignableFrom(sourceType)
-					|| Locale.class.isAssignableFrom(sourceType)) {
-				// TODO use jackson serializer here
-				plainString = "\"" + plainString.replaceAll("\"", "\\\"") + "\"";
+					|| sourceType == char.class || sourceType.isEnum() || Locale.class.isAssignableFrom(sourceType)) {
+				try {
+					plainString = objectMapper.writeValueAsString(plainString);// put quotes around strings
+				} catch (JsonProcessingException e) {
+					throw new RuntimeException(e);
+				}
 			}
 			plainText = plainString.getBytes(StandardCharsets.UTF_8);
 		} else { // an entity
-			plainText = JsonObject.fromJson(context.read(value).toString().getBytes(StandardCharsets.UTF_8)).toBytes();
+			CouchbaseDocument doc = new CouchbaseDocument();
+			context.getConverter().writeInternalRoot(value, doc, property.getTypeInformation(), false, property, false);
+			plainText = JsonObject.from(doc.export()).toBytes();
 		}
```

Comment:
The old code relied on the entity (Airport, Address etc) produced by context.read(value) have a nice toString() method that gives the json representation of the object.  It's a bad assumption.